src/backend/distributed/shared_library_init.c

```diff
@@ -678,6 +678,9 @@ StartupCitusBackend(void)
 	 * the gpid from the application_name.
 	 */
 	AssignGlobalPID();
+
+	SetBackendDataAfterAuthData();
```

Comment:
AssignGlobalPID is idempotent whereas SetBackendDataAfterAuthData is not, so this is a slight behavioural change (at least from main). This may cause issues since this function can be called many times, and the results of GetUserId may differ.

Reply:
I removed userId from the backenddata to resolve this.